sympy/core/tests/test_numbers.py

```diff
@@ -2002,7 +2002,7 @@ def test_issue_10020():
 def test_invert_numbers():
     assert S(2).invert(5) == 3
     assert S(2).invert(Rational(5, 2)) == S.Half
-    assert S(2).invert(5.) == 0.5
```

Comment:
Is `5.1` not a float here? Maybe because of sympify/python? It appears to me as if this test was meant to return a float, but perhaps it stopped doing so during some commit?

Reply:
I'm not sure if this is something that was changed at some point or if the original author simply didn't check.

Reply:
The code that handles this is here:
https://github.com/sympy/sympy/blob/857a97e66c6bb162eb922179aea20226c7a0ed3b/sympy/core/numbers.py#L546-L550
I have no idea what it is trying to do but it ignores the type of `m` if `m > 1`. Hence:
```python
In [1]: S(2).invert(5.)
Out[1]: 1/2

In [2]: S(2.).invert(5.)
Out[2]: 0.500000000000000

In [3]: S(2.).invert(5)
Out[3]: 0.500000000000000
```
If both are integers:
```python
In [4]: S(2).invert(5)
Out[4]: 3
```